lib/Factory.js

```diff
@@ -28,7 +28,8 @@ class Factory {
     // for hana client
     return new Promise((resolve, reject) => {
       const conn = hana.createConnection();
-      conn.connect(dbParams, err => err ? reject(err) : resolve(conn));
+      conn.connect(dbParams, err => {
+          conn.connect(dbParams, err => err ? reject(JSON.stringify(err)) : resolve(conn));
```

Comment:
Thanks for putting err to JSON.stringify(err). While may I know whether there is any specific reason for the retry?

Reply:
I had trouble logging the error, as only Object object went through. Somewhere after this code there is a part which will call toString() on the object.

For example [here](https://github.com/ckyycc/hdb-pool/blob/master/lib/PoolManager.js#L36), but probably more places aswell

Reply:
I mean why we need to retry (conn.connect) when the error occurs, is there any specific reason for the retry?  JSON.stringify(err)  part is good to me.

Reply:
sorry, it was a copy paste error, my local modification was without the retry...